tests/components/border_editor.test.ts

```diff
@@ -0,0 +1,229 @@
+}
+
+beforeEach(() => {
+  fixture = makeTestFixture();
+});
+
+afterEach(() => {
+  fixture.remove();
+});
+
+describe("Can set borders", () => {
+  let model: Model;
+  beforeEach(async () => {
+    model = new Model();
+    await mountBorderEditor({}, model);
+  });
+
+  test("Can set all borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("all");
+
+    for (const cell of ["A1", "A2", "B1", "B2"]) {
+      expect(getBorder(model, cell)).toEqual({
+        top: DEFAULT_BORDER_DESC,
+        bottom: DEFAULT_BORDER_DESC,
+        left: DEFAULT_BORDER_DESC,
+        right: DEFAULT_BORDER_DESC,
+      });
+    }
+  });
+
+  test("Can set only left borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("left");
+
+    expect(getBorder(model, "A1")).toEqual({ left: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "A2")).toEqual({ left: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "B1")).toBeNull();
+    expect(getBorder(model, "B2")).toBeNull();
+  });
+
+  test("Can set only right borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("right");
+
+    expect(getBorder(model, "B1")).toEqual({ right: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "B2")).toEqual({ right: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "A1")).toBeNull();
+    expect(getBorder(model, "A2")).toBeNull();
+  });
+
+  test("Can set only top borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("top");
+
+    expect(getBorder(model, "A1")).toEqual({ top: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "B1")).toEqual({ top: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "A2")).toBeNull();
+    expect(getBorder(model, "B2")).toBeNull();
+  });
+
+  test("Can set only bottom borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("bottom");
+
+    expect(getBorder(model, "A2")).toEqual({ bottom: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "B2")).toEqual({ bottom: DEFAULT_BORDER_DESC });
+    expect(getBorder(model, "A1")).toBeNull();
+    expect(getBorder(model, "B1")).toBeNull();
+  });
+
+  test("Can set only inside hv borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("hv");
+
+    expect(getBorder(model, "A1")).toEqual({
+      right: DEFAULT_BORDER_DESC,
+      bottom: DEFAULT_BORDER_DESC,
+    });
+    expect(getBorder(model, "A2")).toEqual({
+      right: DEFAULT_BORDER_DESC,
+      top: DEFAULT_BORDER_DESC,
+    });
+    expect(getBorder(model, "B1")).toEqual({
+      left: DEFAULT_BORDER_DESC,
+      bottom: DEFAULT_BORDER_DESC,
+    });
+    expect(getBorder(model, "B2")).toEqual({
+      left: DEFAULT_BORDER_DESC,
+      top: DEFAULT_BORDER_DESC,
+    });
+  });
+
+  test("Can set only inside hv borders", async () => {
+    setSelection(model, ["A1:B2"]);
+    await setDefaultBorder("h");
```

Comment:
test title doesn't match the test (same for a few tests below)
```suggestion
  test("Can set only inside horizontal borders", async () => {
    setSelection(model, ["A1:B2"]);
    await setDefaultBorder("h");
```